File: problems/2347_best_poker_hand/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    string bestHand(vector<int>& ranks, vector<char>& suits) {
        if (set<char>(suits.begin(), suits.end()).size() == 1) return "Flush";
        map<int, int> count;
        for (int r : ranks) count[r]++;
        int mx = 0;
        for (auto& [k, v] : count) mx = max(mx, v);
        if (mx >= 3) return "Three of a Kind";
        if (mx == 2) return "Pair";
        return "High Card";
    }
};

File: problems/2347_best_poker_hand/solution.py
from typing import List


class Solution:
    def bestHand(self, ranks: List[int], suits: List[str]) -> str:
        if len(set(suits)) == 1:
            return 'Flush'
        count = Counter(ranks)
        mx = max(count.values())
        if mx >= 3: return 'Three of a Kind'
        if mx == 2: return 'Pair'
        return 'High Card'
